ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -926,3 +931,29 @@ func (r *RayServiceReconciler) labelHealthyServePods(ctx context.Context, rayClu
 
 	return nil
 }
+
+func generateRayClusterJsonHash(rayClusterSpec rayv1alpha1.RayClusterSpec) (string, error) {
+	// Mute all fields that will not trigger new RayCluster preparation. For example,
+	// Autoscaler will update `Replicas` and `WorkersToDelete` when scaling up/down.
+	updatedRayClusterSpec := rayClusterSpec.DeepCopy()
+	for i := 0; i < len(updatedRayClusterSpec.WorkerGroupSpecs); i++ {
+		updatedRayClusterSpec.WorkerGroupSpecs[i].Replicas = nil
```

Comment:
Nice! Is there any way to guard against new fields that we should also ignore? Right now, we're specifically muting `Replicas` and `ScaleStrategy.WorkersToDelete`. My concern is that we add a new field to `RayClusterSpec` that we should also, but we forget to mute it here.

This is a minor suggestion, so if you feels it's not worth the refactoring, please feel free to ignore!

Reply:
Good question, not sure whether a blacklist or whitelist approach is better here and I don't think there's a clear winner.  Tentatively I'd say the blacklist approach (current approach) is better because we expect that a typical new field should trigger a restart.

Reply:
I vote for the blacklist approach (current approach). The blacklist includes both Case 1 and Case 2. Case 1 fields should include those that are fully managed by Ray (`WorkersToDelete` seems to be the only one I can come up with at this moment.). For Case 2 fields, we should ensure that they are supported by the RayCluster rolling update. Currently, only the `Replicas` field for workers can be updated after the RayCluster is created.

Hence, most new fields will not belong to Case 1 and Case 2, so the blacklist can keep short.

> Case 1. RayService should ignore this update and does nothing. (e.g. WorkersToDelete)
Case 2. RayService should not create a new RayCluster, but should update RayCluster (e.g. update RayCluster's labels)
Case 3. RayService should trigger a new RayCluster creation (e.g. update the RayCluster's Docker image)

Reply:
That sounds good, I'll resolve this comment.